src/gui/editors/SongEditor.cpp

```diff
@@ -531,41 +541,88 @@ void SongEditor::keyPressEvent( QKeyEvent * ke )
 }
 
 
-
-
 void SongEditor::wheelEvent( QWheelEvent * we )
 {
-	if( we->modifiers() & Qt::ControlModifier )
+	if ((we->modifiers() & Qt::ControlModifier) && (position(we).x() > m_trackHeadWidth))
 	{
-		int z = m_zoomingModel->value();
+		// calculate the min zoom based on song length and song editor size
+		int minZoom = calculateMinZoom();
+				
+		int z = m_zoomingLogModel->value();
+		if (we->angleDelta().y() > 0)
+			z = static_cast<int>(
+				std::ceil((std::floor(z / 100.0f * DEFAULT_PIXELS_PER_BAR) + 1) * 100.0f / DEFAULT_PIXELS_PER_BAR));
```

Comment:
z += std::max(1.0, z * 0.05)

or something like that has a nicer feeling IMO.

Reply:
Unfortunately, I think it's has to be more complicated than that because zoom is in pixels and sometimes z*0.05 returns a value that doesn't increase the zoom at all.